Source/DafnyLanguageServer.Test/Diagnostics/SimpleLinearVerificationDiagnosticTester.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.Dafny.LanguageServer.IntegrationTest.Extensions;
+using Microsoft.Dafny.LanguageServer.IntegrationTest.Util;
+using Microsoft.Dafny.LanguageServer.Workspace.Notifications;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Microsoft.Dafny.LanguageServer.IntegrationTest.Diagnostics;
+
+[TestClass]
+public class SimpleLinearVerificationDiagnosticTester : LinearVerificationDiagnosticTester {
+  private const int MaxTestExecutionTimeMs = 10000;
+
+  // To add a new test, just call VerifyTrace on a given program,
+  // the test will fail and give the correct output that can be use for the test
+  // Add '//Next<n>:' to edit a line multiple times
+
+  [TestMethod, Timeout(MaxTestExecutionTimeMs)]
+  public async Task EnsureVerificationDiagnosticsAreWorking() {
+    await VerifyTrace(@"
+ .  |  |  |  I  I  |  | :predicate Ok() {
+ .  |  |  |  I  I  |  | :  true
+ .  |  |  |  I  I  |  | :}
+    |  |  |  I  I  |  | :
+ .  S [S][ ][I][I][S] | :method Test(x: bool) returns (i: int)
+ .  S [=][=][-][-][~] | :   ensures i == 2
+ .  S [S][ ][I][I][S] | :{
+ .  S [S][ ][I][I][S] | :  if x {
+ .  S [S][ ][I][I][S] | :    i := 2;
+ .  S [=][=][-][-][~] | :  } else {
```

Comment:
Yes it is useful, and here is the subtle difference.
- Column 6 marks that verification is obsolete. meaning Dafny acknowledge its verification status is outdated. Not only it's useful in "verify on save", to indicate that the user will need to save to see more up-to-date verification information, but it's also useful in "verify on change", because it indicates that the verifier did not start handling this range yet.
- Column 7 marks that there is a process currently working on verifying the range, so the "verification time" can be grasped by the time between column 7 and column 8, which is what users care the most.
If we had only one column, for very long files it would feel like some methods are extremely slow to verify, while others verify very fast, and it does not matches both the mental model (users know that methods are verified one by one) nor the underlying verification model (we split into implementations and verify them one by one, depending on the number of available cores)

In practice, once [this](https://github.com/dafny-lang/dafny/pull/1944) will be merged, the column 6 will not be visible because verification will start immediately.